src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -752,11 +751,31 @@ public SymbolComparer(MatchSymbols matcher)
                     _matcher = matcher;
                 }
 
-                public bool Equals(Symbol x, Symbol y)
+                public bool Equals(Symbol left, Symbol right)
                 {
-                    var other = _matcher.Visit(x);
-                    Debug.Assert((object)other != null);
-                    return other == y;
+                    if (left.Kind != right.Kind)
+                    {
+                        return false;
+                    }
+
+                    var anonymousLeft = left as AnonymousTypeManager.AnonymousTypePublicSymbol;
+                    if ((object)anonymousLeft != null)
+                    {
+                        left = AnonymousTypeManager.TranslateAnonymousTypeSymbol(anonymousLeft);
+                    }
+
+                    var anonymousRight = right as AnonymousTypeManager.AnonymousTypePublicSymbol;
+                    if ((object)anonymousRight != null)
+                    {
+                        right = AnonymousTypeManager.TranslateAnonymousTypeSymbol(anonymousRight);
+                    }
+
+                    if ((object)anonymousLeft != null || (object)anonymousRight != null)
```

Comment:
Would this get any simpler extracting a helper method?
if (TranslateIfAnonymousTypePublicSymbol(ref left) || TranslateIfAnonymousTypePublicSymbol(ref right)) { ... }